File: trunk/x10.lib/src/clock.cc
/*
 * (c) Copyright IBM Corporation 2007
 *
 * $Id: clock.cc,v 1.3 2007-12-10 09:29:57 srkodali Exp $
 * This file is part of X10 Runtime System.
 */

/** Implementation file for (DUMMY) Clock interface. **/

#include <x10/clock.h>
#include <x10/xmacros.h>
#include <lapi.h>
#include "x10libP.h"

namespace x10lib {

/* Perform next operation on the specified clock. */
x10_err_t x10lib::ClockNext(Clock *c, int n)
{
	// refine???
	LRC(LAPI_Gfence(__x10_hndl));
	return X10_OK;
}

} /* closing brace for namespace x10lib */

File: x10.runtime.17/src/x10/runtime/Clock.java
/*
 *
 * (C) Copyright IBM Corporation 2006-2008
 *
 *  This file is part of X10 Language.
 *
 */
package x10.runtime;

/**
 * Used to access XRX clock implementation from Java runtime
 * 
 * @author tardieu
 *
 */
public interface Clock {
	public void _register();
	public void _resume();
	public void _next();
	public void _drop();
}
